app/src/main/java/com/duckduckgo/app/browser/NestedWebView.kt

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2017 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser
+
+import android.content.Context
+import android.support.v4.view.NestedScrollingChildHelper
+import android.support.v4.view.ViewCompat
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.webkit.WebView
+
+/**
+ * NestedWebView which allows the WebView to hide the toolbar when placed in a CoordinatorLayout
+ *
+ * Based on https://github.com/takahirom/webview-in-coordinatorlayout
+ */
+class NestedWebView : WebView {
+
+    constructor(context: Context) : this(context, null)
+    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)
+
+    private var lastY: Int = 0
+    private val scrollOffset = IntArray(2)
+    private val scrollConsumed = IntArray(2)
+    private var nestedOffsetY: Int = 0
+
+    private val mChildHelper: NestedScrollingChildHelper = NestedScrollingChildHelper(this)
```

Comment:
Are we planning to maintain this as our own over time?  If so can we use our own style guide e.g remove the `m` prefix from properties. If not, I'd suggest putting it in the a third party package to hide it away.

Reply:
Updated to use our own style.

I'm not sure if this class will be needed longer term. I suspect it'll either disappear completely, or have more work done to it. Don't think it'll live forever as is, so have tidied up a little.